client/ayon_core/hosts/houdini/api/plugin.py

```diff
@@ -253,7 +255,15 @@ def update_instances(self, update_list):
                 key: changes[key].new_value
                 for key in changes.changed_keys
             }
-            # Update parm templates and values
+
+            # Update Houdini node's parameters based on creator attributes.
+            # This is done by re-using the logic inside the creators.
+            host = registered_host()
+            create_context = CreateContext(host, reset=True)
+            create_plugin = create_context.creators[created_inst.creator_identifier]
+            create_plugin.update_node_parameters(instance_node, new_values["creator_attributes"])
```

Comment:
`update_instances` should **NOT** create a new `CreateContext`

You are trying to get the Create plug-in. But that is the **current** plug-in instance!
This should just be:

```suggestion
            self.update_node_parameters(instance_node, new_values["creator_attributes"])
```

Also, I wouldn't limit this to `creator_attributes` but just pass it the full `changes`. Allow any subclass to do with that what they want.

Reply:
You seem to be lacking logic that does the reverse of updating the node parameters. The creator attributes (or whatever other data you are persisting on the node) should likely also be read from the node again? So that if the user had toggled that on the node itself that the publisher UI also reads that as the ground truth for the creator attribute, etc.

Reply:
using the rop node as the source of truth is a crucial part in Houdini addon. 
Maybe I was confused when I used `get_instance_attr_defs` for the first time in the other PR.

So, my point here https://github.com/ynput/ayon-core/pull/328/files/a894a9a0897ecc82460af92064527088844409cd#r1558203127 was wrong and I really don't need to update the node state/parameters to align with user preference before the validation phase as collectors should collect/deduce the data?

Reply:
Maybe I'm overthinking. 
But, when applying changes in the publish tab I expected _as a user_ that clicking save button will apply these changes.
Also, knowing the fact that eventually the ROP node will follow the settings in the publish tab makes me willing to use the publisher UI more than doing anything manually in the rop node as the publisher **knows** how to change the parameters properly. 

But, it could be only me.

Reply:
I agree with @BigRoy. We shouldn't create context when we are updating the instances but instead, we need to find AYON-related parameters which already existed in the node and update them. If I am not wrong, `CreateContext` at this point can probably add unnecessary data within the existing data. This would cause issues when the publisher trying to collect instance data from the instance node(it would probably recognize multiple instances where the un-updated and updated instances live together in the publisher UI).

Reply:
> Save button in publisher should only update AYON-related parameters.

This isn't entirely a hard rule. There may be reasons to want a native Houdini parameter to be used as the value for a setting exposed in the publisher UI - but I'd also then try to "read" the state of that publisher value from that parameter instead of storing other 'metadata' on the node that gets out of sync.

> Publish instances (Houdini ROPs) should be the source of truth.

As much as possible, yes.

> Collector plugins should collect/deduce/predict the data according to the ROP node state and user settings in the publisher

If we can provide a workflow to the user that is non-destructive and works for our pipeline needs then I think that can be very nice. With Houdini being quite technical software where users can do very advanced stuff there is a lot of merit in staying "thin" on top of Houdini and just trying to allow as many stuff through by the artists (they are in control).

It takes away a bit from being able to automate _everything_ via solely AYON logic -but it does mean things remain flexible for Houdini artists.

There will still be reasons to hijack Houdini and take control ourselves from AYON if we really need to. But if we can avoid it, then I think it's good.

> Extract plugins can be used to alter the state of the node (e.g. Align the node parameters to the instance options in the publisher)

That's totally fine, yes - preferred actually. **BUT** it MUST return it to its original state afterwards. The extractor is allowed to do anything - but at the end of it leave the scene unchanged.

The same is true basically for any pyblish plug-in. If a validator requires putting down an extra node in the node tree, run the node, get the result - just to do the validation, then by all means implement it. As long as at the end of the `process()` method the scene is back exactly how it was before it. Not saying it's good practice to do this heavy stuff, but it's totally fine to do whatever you want within a Pyblish plug-in like that.

> Therefore this PR should be closed.
> What do you think ?

Going from my points above, I can't say a hard yes. I think it's actually **no** - there's merit to this PR. But it totally depends on what it is trying to solve.

If the extractor works fine by changing things 'temporarily' and extracting it - then that's much better than updating the user's scene from the creator I think. However, that logic may not work if e.g. the scene is to be submitted to Deadline and the Deadline job will do whatever the ROP node is configured to. In that case we can't really 'temporarily hijack' the scene state just for the render. As such, for those cases the scene _must_ be in the correct state for the Deadline Job to do what we want it to do. In that case, having it update from the Creator and (also read the state from the node's parameters) could make a much better user experience instead of relying on Validators with Repair functionality.

Reply:
>If the extractor works fine by changing things 'temporarily' and extracting it - then that's much better than updating the user's scene from the creator I think.

maybe the `imprint` function in `lib` would help for this.
https://github.com/ynput/ayon-core/blob/9299e7e7033da85b26e464494efa15ea8e53a888/client/ayon_core/hosts/houdini/api/lib.py#L259

Reply:
I've applied your suggested code.. also, I added a note about it (tbh, I didn't the type of self and it's indeed not needed to create another object) Thanks! 

https://github.com/ynput/ayon-core/pull/418/commits/eda3b88e7c6eed9321196c93ce729a732fa7e3a9